### PR TITLE
159991039 logout popup

### PR DESCRIPTION
**Pull request by lvonah**
https://www.pivotaltracker.com/story/show/159991039
Team Common/Auth - Updating logout functionality to match login functionality using a logout popup.

Instructions to test:
Pull code
RUN "npm install"
RUN "npm install -g @angular/cli@6.0.7" if you don't have current global angular CLI
RUN "ng serve"
Open browser to localhost:4200
Login using our credentials. Username: lvonah@project3auth.onmicrosoft.com Password: Revature!

### CODE REVIEW

File: src/app/components/navbar/navbar.component.html [hunk 1]
```diff
@@ -1,20 +1,48 @@
...
     </mat-grid-tile>
+
+    <!-- This tile displays the user's firstName using interpolation. -->
     <mat-grid-tile class="nav-tile">
       <div id="user" *ngIf="LoggedIn">WELCOME, {{firstName[0] | uppercase}}</div>
     </mat-grid-tile>
+
+    <!--
+      For large screen sizes:
+      This tile changes to the appropriate login or logout button while loading property is false;
```
Comment (Tallisble) [inline src/app/components/navbar/navbar.component.html]: change ";" to "."

File: src/app/components/navbar/navbar.component.html [hunk 2]
```diff
@@ -24,16 +52,32 @@
         </button>
       </div>
     </mat-grid-tile>
+
```
Comment (Tallisble) [inline src/app/components/navbar/navbar.component.html]: remove space

File: src/app/services/auth.service.ts [hunk 3]
```diff
@@ -18,15 +18,23 @@ export class AuthService {
...
+     * isLoggedIn() function leverages MSAL's UserAgentApplication to access their getUser() function.
+     * Returns true if user is found, otherwise false.
+     */
     get isLoggedIn() {
         const check: boolean = this.app.getUser();
         if (check) {
             return true;
         } else {
             return false;
         }
```
Comment (escalonn) [inline src/app/services/auth.service.ts]: just return `this.app.getUser() === null` or `this.userData === null`

File: src/app/services/auth.service.ts [hunk 4]
```diff
@@ -35,8 +43,14 @@ export class AuthService {
             return null;
         }
     }
```
Comment (escalonn) [inline src/app/services/auth.service.ts]: just return `this.app.getUser()`... the only way user would be falsy is if it's null

File: src/app/services/auth.service.ts [hunk 5]
```diff
@@ -35,8 +43,14 @@ export class AuthService {
...
+     * Login function leveraging MSAL's UserAgentApplication to access their loginPopup() function.
+     * This adds a promise that checks for a user with the getUser() function from UserAgentApplication.
+     * Returns user if found, otherwise null.
+     */
     public login() {
-        return this.app.loginPopup(this.applicationConfig.graphScopes,  this.applicationConfig.extraQueryParameter)
+        return this.app.loginPopup(this.applicationConfig.graphScopes, this.applicationConfig.extraQueryParameter)
             .then(idToken => {
                 const user = this.app.getUser();
                 if (user) {
```
Comment (escalonn) [inline src/app/services/auth.service.ts]: just have `idToken => this.app.getUser()` or `idToken => this.userData`

File: src/app/services/auth.service.ts [hunk 6]
```diff
@@ -48,9 +62,28 @@ export class AuthService {
                 return null;
             });
```
Comment (escalonn) [inline src/app/services/auth.service.ts]: `() => null`

**Pull request merged**